[#2845] Display temp max on character sheet HP bar

arbron opened the pull request:
<img width="243" alt="Screenshot 2024-02-08 at 10 34 43" src="https://github.com/foundryvtt/dnd5e/assets/19979839/b829cf0d-6080-4395-b86a-1f6b2157c0ed">

Closes #2845

Fyorl commented:
I think we should show the total hit points on the meter rather than having the base hit points with a (+/- X). Part of the advantage of using a VTT is that it does these sorts of calculations for you, so that you can see exactly how many max hit points your character currently has. If we have an ActiveEffect adding +5 to an ability score, we don't display it as 15 (+5), for example.

If we're worried about people forgetting they have a temporary adjustment applied, I might suggest something like this which shows a bonus indicator. There's the danger that a user thinks they have 25 HP + another 5 of course, so it's not perfect, but the best I've been able to come up with for now. Any further suggestions/iteration are welcome.

![image](https://github.com/foundryvtt/dnd5e/assets/207433/d98babe5-7550-49b0-908d-0fa211006fcc)

We add an additional `<span class="bonus">+5</span>` and the styles are just:
```css
.dnd5e2 .meter .label {
  color: var(--color-text-light-0);
  position: relative;
  padding: 0 0.75rem 0 0.5rem;
  display: flex;
  gap: 0.5rem;
  width: 100%;
  align-items: center;
}

.dnd5e2 .meter .label .base {
  margin-left: auto;
  font-size: var(--font-size-14);
  opacity: .8;
}
```

krbz999 commented:
> Any further suggestions/iteration are welcome.

When affected by `hp.tempmax`, additionally changing the maximum to a blueish color could help indicate that it is not the default maximum. (In addition to the inserted span.)

arbron commented:
Restyled and added `hp.effectiveMax` to avoid having to recalculate that value throughout the code like we were before.

Modified the color based on whether the temp is positive or negative, here's how that looks:

<img width="239" alt="Screenshot 2024-02-09 at 09 48 55" src="https://github.com/foundryvtt/dnd5e/assets/19979839/e9a1e99a-f160-4059-a1d6-32220608250e">
<img width="235" alt="Screenshot 2024-02-09 at 09 49 04" src="https://github.com/foundryvtt/dnd5e/assets/19979839/e4b5f50b-38cd-4bd1-bb9a-189f55fe73d7">
